### PR TITLE
Fixed some bugs and usage issues detected by Clang-tidy.

### PR DESCRIPTION
**Pull request by monadbobo**
depend #214, and when #214 is merged, the clang-tidy will be added to the CI.

### CODE REVIEW

File: src/common/base/test/EitherOrTest.cpp [hunk 1]
```diff
@@ -228,29 +228,25 @@ TEST(EitherOr, MoveConstructFromValue) {
     {
         EitherOr<int, std::string> result1(101);
         auto result2 = std::move(result1);
-        ASSERT_TRUE(result1.isVoid());
```
Comment (Shylock-Hg) [inline src/common/base/test/EitherOrTest.cpp:231]: Why remove this?

File: src/common/clients/storage/GraphStorageClient.cpp [hunk 2]
```diff
@@ -10,22 +10,22 @@
...
-                                 const std::vector<cpp2::Expr>* expressions,
-                                 bool dedup,
-                                 bool random,
-                                 const std::vector<cpp2::OrderBy>& orderBy,
-                                 int64_t limit,
-                                 std::string filter,
-                                 folly::EventBase* evb) {
+folly::SemiFuture<StorageRpcResponse<cpp2::GetNeighborsResponse>> GraphStorageClient::getNeighbors(
+    GraphSpaceID space,
+    const std::vector<std::string>& colNames,
```
Comment (dutor) [inline src/common/clients/storage/GraphStorageClient.cpp:15]: Such changes are not appropriate.

File: src/common/base/test/EitherOrTest.cpp (hunk 1 above)
Comment (monadbobo) [reply src/common/base/test/EitherOrTest.cpp:231]: You can't use a value that has been moved.

File: src/common/clients/storage/GraphStorageClient.cpp (hunk 2 above)
Comment (yixinglu) [reply src/common/clients/storage/GraphStorageClient.cpp:15]: @monadbobo you can add `NOLINT` in the returned value line to tell clang-tidy not to format this function signature.

**Pull request closed without merging**